tutor/src/store/courses.ts

```diff
@@ -0,0 +1,46 @@
+import { createSlice, createEntityAdapter, PayloadAction, current } from '@reduxjs/toolkit'
+import { useSelector } from 'react-redux'
+import { Course } from './types'
+import { updateCourse } from './api'
+import { bootstrap } from './bootstrap'
+
+const courseAdapter = createEntityAdapter<Course>({
+    sortComparer: (a, b) => a.name.localeCompare(b.name),
+})
+
+const initialState = courseAdapter.getInitialState()
+interface CourseSlice { courses: typeof initialState }
+
+const selectors = courseAdapter.getSelectors<CourseSlice>(s => s.courses)
```

Comment:
selectors have a bunch of nice functions: https://redux-toolkit.js.org/api/createEntityAdapter#return-value